### PR TITLE
Add new api for signal and string

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
1. support strdupa and strndupa api.
2. support sigisemptyset, sigandset, sigorset api.
## Impact
support new api
## Testing
local test.

### CODE REVIEW

**Comment by davids5** [comment]
@Donny9 - Could this be separate PRs?

**Comment by xiaoxiang781216** [comment]
> @Donny9 - Could this be separate PRs?

@davids5 it's already separated into two patch one for signal and one for string. The change is very simple, does we really need split the simple change into two PR?

File: include/string.h [hunk 1]
```diff
@@ -35,7 +35,9 @@
  ****************************************************************************/
 
 #define strcoll_l(s1, s2, l)    strcoll(s1, s2)
+#define strdupa(x)              strcpy(alloca(strlen(x) + 1), x)
 #define strerror_l(e, l)        strerror(e)
+#define strndupa(x, len)        strncpy(alloca(strlen(x) + 1), x, len)
```
Comment (xiaoxiang781216) [inline include/string.h:40]: The implementation looks wrong:
https://github.com/esmil/musl/blob/master/include/string.h#L89-L95

Comment (Donny9) [reply include/string.h]: Done!

File: include/string.h [hunk 2]
```diff
@@ -35,8 +35,16 @@
  ****************************************************************************/
 
 #define strcoll_l(s1, s2, l)    strcoll(s1, s2)
+#define strdupa(x)              strcpy(alloca(strlen(x) + 1), x)
 #define strerror_l(e, l)        strerror(e)
 #define strxfrm_l(s1, s2, n, l) strxfrm(s1, s2, n)
+#define	strndupa(x, len) ({ \
```
Comment (xiaoxiang781216) [inline include/string.h:41]: ({ is gcc extension, let's change to c function?

Comment (Donny9) [reply include/string.h]: Done!

Comment (Donny9) [reply include/string.h]: keep, strndupa return alloca local memory, so using marco to do it.

Comment (xiaoxiang781216) [reply include/string.h]: Ok

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**